posthog/api/test/test_prompt.py

```diff
@@ -0,0 +1,181 @@
+        key="activation-checklist",
+        type="one-off",
+        path_match=["/*"],
+        path_exclude=["/ingestion", "/ingestion/*"],
+        status="active",
+    )
+    sequence2.prompts.add(prompt2)
+    sequence2.must_have_completed.add(sequence1)
+
+
+class TestPrompt(APIBaseTest):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+    def setUp(cls):
+        distinct_id_user = User.objects.create_and_join(cls.organization, "distinct_id_user@posthog.com", None)
+        distinct_id_user.distinct_id = "distinct_id"
+        distinct_id_user.save()
+        cls.user = distinct_id_user
+
+    @freeze_time("2022-08-25T22:09:14.252Z")
+    def test_my_prompts(self):
+        self.client.force_login(self.user)
+        _setup_prompts()
+        # receive only the one sequence which doesn't have prerequisites
+        response = self.client.patch(f"/api/prompts/my_prompts", {}, format="json")
+        assert response.status_code == status.HTTP_200_OK
+        json_response = response.json()
+        assert len(json_response["sequences"]) == 1
+        assert json_response["sequences"][0]["key"] == "start-flow"
+
+        # updates the saved state using the more recent local state
+        local_state = {
+            "start-flow": {
+                "sequence": "start-flow",
+                "last_updated_at": "2022-08-25T22:09:14.252Z",
+                "step": 0,
+                "completed": True,
+                "dismissed": False,
+            }
+        }
+        response = self.client.patch(f"/api/prompts/my_prompts", local_state, format="json")
+        assert response.status_code == status.HTTP_200_OK
+        json_response = response.json()
+        # we now also receive the other sequences
+        assert len(json_response["sequences"]) == 2
+        assert json_response["state"]["start-flow"]["step"] == 0
+        assert json_response["state"]["start-flow"]["completed"] is True
+
+        saved_states = list(UserPromptState.objects.filter(user=self.user))
+        assert len(saved_states) == 1
+        first_saved_state = list(saved_states)[0]
+        assert first_saved_state.step == 0
+        assert first_saved_state.completed is True
+
+        # ignores the local state as it is less recent
+        local_state = {
+            "start-flow": {
+                "sequence": "start-flow",
+                "last_updated_at": "2022-08-24T22:09:14.252Z",
+                "step": 1,
+                "completed": False,
+                "dismissed": False,
+            }
+        }
+        response = self.client.patch(f"/api/prompts/my_prompts", local_state, format="json")
+        assert response.status_code == status.HTTP_200_OK
+        json_response = response.json()
+        assert len(json_response["sequences"]) == 2
+        assert json_response["state"]["start-flow"]["step"] == 0
+        assert json_response["state"]["start-flow"]["completed"] is True
+
+    def test_webhook(self):
```

Comment:
This test offers no context behind the assertions (and there's a LOT of them). Can you add some comments for the future traveler (or the present one, like me)? Would also be nice if this weren't as long, but I see only one API request is made, so I understand if this is intended